i2c/24lc32_i2c/24lc32_i2c_lib.c

```diff
@@ -0,0 +1,85 @@
+#include <stdio.h>
+#include "pico/stdlib.h"
+#include "hardware/i2c.h"
+
+// This is the max number of bytes which can be written at once during a page write
+#define MAX_PAGE_WRITE 32
+
+// Acknowledge polling can be used to see when a write cycle is complete
+// Wait for the i2c slave to acknowledge our empty write
+bool ack_poll(uint8_t i2c_addr) {
+    int timeout = 100;
+    for (int i = 0; i < timeout; i++) {
+        if (i2c_write_blocking(i2c_default, i2c_addr, NULL, 1, false) == 1) {
+            return true;
+        }
+        sleep_ms(1);
+    }
+    printf("ack timeout\n");
+    return false;
+}
+
+// Read num bytes of memory starting at given address into result
+void read_eeprom(uint16_t address, uint8_t *result, uint16_t num, uint8_t i2c_addr) {
```

Comment:
Perhaps this should check that `(address + num - 1) < 4096` ? (and similarly in the other read / write functions)
Might also be worth doing e.g. `#define EEPROM_SIZE 4096` ?